packages/website/pages/docs/concepts/creating-l2-blocks/intrinsic-validity-functions.mdx

```diff
@@ -0,0 +1,63 @@
+The Ethereum yellow paper has a well defined set of rules to compute the state transition. We use these same rules to take a proposed block, and compute the post-block state on Taiko. A proposed block on Taiko has two parts:
+
+- The block metadata
+- A list of transactions (stored in a blob, and the block metadata stores the hash to this blob)
+
+We divide the yellow paper validity checks into two parts:
+
+- **Proposed block** intrinsic validity function
+- **Transaction list** intrinsic validity function
+
+A proposed block must pass these two checks in order to construct an L2 block on Taiko. If a block passes the proposed block validity function but later fails the transaction list validity function, it will be proven as invalid.
+
+## Proposed block intrinsic validity function
+
+The formal specification of the proposed block intrinsic can be found in the [whitepaper](/resources/whitepaper), but on a high level, the proposed block intrinsic validity function checks that the block is valid except for transaction list validity checks. This is separated because on TaikoL1, the transaction list is not known and stored as a data blob. It would also be too expensive to do these checks on L1.
+
+| Name                    | Symbol            | Meaning                                                                              |
+| ----------------------- | ----------------- | ------------------------------------------------------------------------------------ |
+| **Block Metadata**      | $C$               | The block metadata.                                                                  |
+| **id**                  | $C_i$             | A value equal to the number of proposed blocks. The genesis block has an id of zero. |
+| **beneficiary**         | $C_c$             | The 20-byte address to which all transaction fees in the block will be transferred.  |
+| **timestamp**           | $C_s$             | The timestamp used in the block, set to the enclosing L1 timestamp.                  |
+| **mixHash**             | $C_m$             | The mixHash value used in the block, set to the enclosing L1 mixHash.                |
+| **txListHash**          | $C_t$             | The Keccak-256 hash of this block’s txList (KZG commitment after EIP-4844).          |
+| **l1Height**            | $C_a$             | The enclosing L1 block’s parent block number.                                        |
+| **l1Hash**              | $C_h$             | The enclosing L1 block’s parent block hash.                                          |
+| **TaikoL1**             | $R$               | The TaikoL1 contract.                                                                |
+| **numProposedBlocks**   | $R_i$             | The id for the next proposed block.                                                  |
```

Comment:
Seems the id for the next proposed block should be `number of proposed blocks` + `1`?

Reply:
[5dc2802](https://github.com/taikoxyz/taiko-mono/pull/13316/commits/5dc2802d8ee58150d79f8027abe2539e6dd532b9)